docs/router.md

```diff
@@ -0,0 +1,216 @@
+# About Routers
+
+`gpii.express.router` is a subclass of [`gpii.express.middleware`](middleware.md) that wraps
+[the  Express `router` object](http://expressjs.com/en/4x/api.html#router), which is simply a container for a collection
+of [middleware](middleware.md).
+
+Both Express and the Express `router` object have the concept of a "stack" of middleware.  A `gpii.express.router`
```

Comment:
I want to leave a reference to `stack` in there if possible.  I've added a block expanding on the previous explanation, which includes:

1. The "tree" wording
2. A link to the Wikipedia article we discussed.
3. The diagram from the Wikipedia article that illustrates the "pre-order" strategy used.
4. A sample component options block that matches the diagram.

Please review, hopefully this is not far from something we can both live with.